l1/queue.go

```diff
@@ -0,0 +1,63 @@
+package l1
+
+import (
+	"github.com/NethermindEth/juno/l1/internal/contract"
+)
+
+// queue is a priority queue that implements a "confirmation window."
+// Given an L1 height, all [contract.LogStateUpdate]s emitted at least
+// confirmationPeriod blocks before the height are "confirmed".
+//
+// Its abnormal API reflects the fact that it is only meant to be used
+// by the L1 verifier.
+type queue struct {
+	confirmationPeriod uint64
+	q                  map[uint64]*contract.StarknetLogStateUpdate
+}
+
+func newQueue(confirmationPeriod uint64) *queue {
+	return &queue{
+		confirmationPeriod: confirmationPeriod,
+		q:                  make(map[uint64]*contract.StarknetLogStateUpdate),
+	}
+}
+
+func (q *queue) Enqueue(update *contract.StarknetLogStateUpdate) {
+	q.q[update.Raw.BlockNumber] = update
+}
+
+func (q *queue) Remove(l1BlockNumber uint64) {
+	delete(q.q, l1BlockNumber)
+}
+
+func (q *queue) Reorg(forkedBlockNumber uint64) {
+	for l1BlockNumber := range q.q {
```

Comment:
loop over map in Go is randomized, can it lead to some issues here?

Reply:
I don't think the order matters here since we loop over the entire map.